plugin/requirejs.js

```diff
@@ -255,19 +255,26 @@
   });
 
   function findTypeAt(_file, _pos, expr, type) {
-    if (!expr || expr.node.type != "Literal" ||
-        typeof expr.node.value != "string" || !expr.node.required)
-      return type;
-
-    // The `type` is a value shared for all string literals.
-    // We must create a copy before modifying `origin` and `originNode`.
-    // Otherwise all string literals would point to the last jump location
-    type = Object.create(type);
-
-    // Provide a custom origin location pointing to the require()d file
-    var exportedType = expr.node.required;
-    type.origin = exportedType.origin;
-    type.originNode = exportedType.originNode;
+    if (!expr) return type;
+    var isStringLiteral = expr.node.type === "Literal" &&
+       typeof expr.node.value === "string";
+    var isRequireArg = !!expr.node.required;
+
+    if (isStringLiteral && isRequireArg) {
```

Comment:
I have copied/pasted the code from node tern plugin and adapt it for requirejs. If you don't like this code, I think we should modify node tern plugin too.